hudi-client/src/main/java/org/apache/hudi/config/HoodieCompactionConfig.java

```diff
@@ -103,6 +104,8 @@
   private static final String DEFAULT_MAX_COMMITS_TO_KEEP = "30";
   private static final String DEFAULT_MIN_COMMITS_TO_KEEP = "20";
   private static final String DEFAULT_COMMITS_ARCHIVAL_BATCH_SIZE = String.valueOf(10);
+  // Do not read more than 6MB at a time (6MB observed p95 in prod from 2 datasets)
+  private static final String DEFAULT_COMMITS_ARCHIVAL_MEM_SIZE = String.valueOf(6 * 1024 * 1024);
```

Comment:
Also, 6MB seems very small, we should increasing this size

Reply:
any suggestions on how to pick this number? I just looked at 2 production dataset and measured size of 10 records in archived file. Max noticed is 10MB. p95 is 6MB.  Let me know if you have suggestions.